### DISCUSSION TITLE
Binary Ninja string analyze problem

### DISCUSSION THREAD

**Discussion by ret2void** [Vector35/binaryninja-api#4065] [General] [answered]
hi, i am trying to decompile a program which includes a chinese string, and it is produced by gcc, so the chinese string would be encoded by utf8。
the string is “中文测试” and in binja it showed as follow:
 
![image](https://user-images.githubusercontent.com/83763549/222702955-28edd930-120b-492b-90e7-c2ab82408398.png)
how can i change it to be a chinese expression? it is a utf-8 string,why binja cannot show the real string?
i have tried a lot of changes,but all of them are invalid:
![image](https://user-images.githubusercontent.com/83763549/222703629-6a8297b6-78e7-4451-ba97-57121e091557.png)
![image](https://user-images.githubusercontent.com/83763549/222703666-5f9575bc-0c52-40a1-a7a8-d9e52ffccdcf.png)
in ghidra, the chinese string appears normal:
![image](https://user-images.githubusercontent.com/83763549/222704037-a9fe9f89-fc88-474f-af01-f2f7bb616bc3.png)

**Comment by ret2void** [comment]
I strongly hope that binja can support utf-8 string
Because this is very important for locating binary key points

**Comment by psifertex** [comment, marked as answer]
So the correct answer is that Binary Ninja has partial support for utf-8 strings they're just slightly more work than ascii strings right now:

Step 1) Use "open with options" and specify the code pages you want: 
<img width="996" alt="Screenshot 2023-03-03 at 6 16 07 AM" src="https://user-images.githubusercontent.com/140215/222706827-379c9942-f4bb-4684-85cc-05af3b07e889.png">
Step 2) Press "a" on the UTF-8 string you want displayed:
Before:
<img width="525" alt="Screenshot 2023-03-03 at 6 12 45 AM" src="https://user-images.githubusercontent.com/140215/222706936-31692047-35f1-4509-99e6-1f7722bc2e51.png">
After:
<img width="396" alt="Screenshot 2023-03-03 at 6 17 20 AM" src="https://user-images.githubusercontent.com/140215/222706998-2a5cbb74-6421-4fca-bba4-a9356ed8ece9.png">

Note that there are still some minor rendering glitches for UTF: 
https://github.com/Vector35/binaryninja-api/issues/1234

And some other related issues would also be helpful:
https://github.com/Vector35/binaryninja-api/issues/2254

**Reply by psifertex** [reply]
Oh, and that rendering glitch in #1234 shows up here in the main view which is why I described the current state as "partial support":
<img width="424" alt="Screenshot 2023-03-03 at 6 27 19 AM" src="https://user-images.githubusercontent.com/140215/222708758-6ca60596-ac5d-4b38-b56d-4e4c7fa74263.png">

**Reply by psifertex** [reply]
The good news is that step 2 won't be required for your binary you show in screenshots since you've already got the proper type information. You should just be able to use open-with-options to specify the codepage and the only issues you'll be running into are the font-width calculation alignment problem shown in my last screenshot.

**Reply by ret2void** [reply]
I got the same answer using your method
Thank you very much and hope Binary Ninja can be more better！
